src/connection_manager.rs

```diff
@@ -590,6 +601,15 @@ fn handle_connect(mut state: WeakState, trans: transport::Transport,
     endpoint
 }
 
+fn increment_bs_count(state: &mut WeakState) -> io::Result<bool> {
+    lock_mut_state(state, move |s: &mut State| {
+        if s.bs_count.0 < s.bs_count.1 {
+            s.bs_count.0 = s.bs_count.0 + 1;
+            return Ok(true);
+        }
+        Ok(false)  // reached max bs connection
```

Comment:
Yes. I am resetting the bootstrap count and all connections whenever we call bootstrap.
https://github.com/maidsafe/crust/pull/253/files#diff-a75471a479fa85679aec0a2b6876bc87R277